astronomer/providers/databricks/operators/databricks.py

```diff
@@ -24,7 +24,12 @@ def execute(self, context: "Context") -> None:
         # Note: This hook makes non-async calls.
         # It is imported from the Databricks base class.
         # Async calls (i.e. polling) are handled in the Trigger.
-        hook = self._get_hook()
+        try:
+            # for apache-airflow-providers-databricks<3.2.0
+            hook = self._get_hook()
+        except TypeError:
+            # for apache-airflow-providers-databricks>=3.2.0
+            hook = self._get_hook(caller="DatabricksSubmitRunOperatorAsync")  # type: ignore[call-arg]
```

Comment:
why do we need `# type: ignore[call-arg]` here?

Reply:
Currently, in rc version, we have the argument `caller` but the  CI job has old version which doesn't know this argument exists so the mypy check fails.

Reply:
CI job should use the new version, isnt it?

Reply:
yes, that doesn't have this arg yet, once the RC gets released and new version is out. Until then mypy will throw an error.

Reply:
so lets introduce a TODO that we need to remove this ignore after the new provider is out

Reply:
Ok

Reply:
added